src/Marten/Events/Projections/Flattened/StatementMap.cs

```diff
@@ -0,0 +1,170 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using Baseline.Reflection;
+using LamarCodeGeneration;
+using Marten.Events.CodeGeneration;
+using Marten.Internal.CodeGeneration;
+using Weasel.Core;
+using Weasel.Postgresql.Tables;
+
+namespace Marten.Events.Projections.Flattened
+{
+    public class StatementMap<T> : IEventHandler
```

Comment:
**SUGGESTION:** I think that it'd be good to add people option to do custom SQL, then someone could do their own magic, and we wouldn't need to add a lot of new weird map permutations immediately.

Reply:
True, but they could also drop into an `EventProjection` and issue all the custom SQL they want. We could definitely do this later though.

Think about this though. The SQL generated by this functionality is doing upserts which are clumsy bits of code to write and would blow up the parameter count if you did it by hand. There is a reason why I went this way.